### ISSUE TITLE
Headings get shown twice in first cell of a notebook

### ISSUE THREAD

**Issue by batpad** [jupyter-book/jupyterlab-myst#229]
## Description

**Steps to reproduce**: In the first cell of a new notebook, convert to a Markdown cell and type the following text:

```
## Test heading

some text
```

**Expected Output**: `Test Heading` to be in an `<h2>` tag or so, and appear once.

**Actual**:  `Test Heading` appears twice. It seems to generate an additional `<h1>` tag with markup like this:

```
<div id="skip-to-frontmatter" aria-label="article frontmatter" class="" style=""><h1 class="mb-0">Test heading</h1></div>
```

Screenshot of what this looks like:

![image](https://github.com/executablebooks/jupyterlab-myst/assets/72280/b0865c5d-471a-4692-9698-68cb604634a8)

Interestingly, this does not affect `<h1>` headings, so this issue does not affect this:

```
# Test heading

some text
```

Also, this only occurs in the first cell of a new notebook and does not affect subsequent cells.

## Additional notes

I am testing this off the `main` branch running Jupyter lab locally. I can also see the issue present on a remote Jupyterhub running `v2.3.1` .

This issue maybe the same as https://github.com/executablebooks/jupyterlab-myst/issues/222#issuecomment-2007702221 ?

cc @rowanc1

**Comment by agoose77** [comment]
Yes, for now this only affects `<h2>` headings. @fwkoch do you have a feeling about whether we want to relax this and lift _any_ heading?

**Comment by batpad** [comment]
gentle bump here .. have had a few more users complain about this .. is there anything we could do to try and help? Thanks again :-)

**Comment by kuhlaid** [comment]
For me if I use the following Markdown:
```
### H3

#### H4
```
...then the output looks like 

## H3
### H3
#### H4

If I use:

```
# H1

## H2
```
...then the output looks normal, like this:

# H1

## H2

It seems that if the highest-tier heading (H1) is not included in the markdown then the duplicate headers occur on the first header listed, regardless of whether the first header is H2, H3, etc.
